src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -695,10 +695,17 @@ Task<bool> IsIntentListenerRegisteredAsync(AppMetadata appMetadata)
 
     private Task RemoveModuleAsync(IModuleInstance instance)
     {
-        var fdc3InstanceId = GetFdc3InstanceId(instance);
-        if (!_runningModules.TryRemove(new(fdc3InstanceId), out _))
+        try
+        {
+            var fdc3InstanceId = GetFdc3InstanceId(instance);
+            if (!_runningModules.TryRemove(new(fdc3InstanceId), out _))
+            {
+                _logger.LogError($"Could not remove the closed window with instanceId: {fdc3InstanceId}.");
+            }
+        }
+        catch (Fdc3DesktopAgentException exception)
         {
-            _logger.LogError($"Could not remove the closed window with instanceId: {fdc3InstanceId}.");
+            _logger.LogError(exception, $"Exception thrown while removing module: {instance.Manifest.Id}, {instance.Manifest.Name} from running instances in FDC3DesktopAgent.");
```

Comment:
Should we be making of use ? in here to guard against nulls in a catch logging statement?

Reply:
Same here, the variables/parameters shouldn't be null values.

Reply:
`RemoveModuleAsync` is also called from a subscribe event, when a module is stopped fired by the moduleLoader itself, so in this case the `IModuleInstance` and it's manifest's id and name shouldn't be null